blockchain/reactor.go

```diff
@@ -137,10 +185,13 @@ func (bcR *BlockchainReactor) respondToPeer(msg *bcBlockRequestMessage, src p2p.
 	block := bcR.store.LoadBlock(msg.Height)
 	if block != nil {
 		msg := &bcBlockResponseMessage{Block: block}
+
+		bcR.MarkPeer(src, true, 1, p2p.Correct)
```

Comment:
lets mark good when we receive a block we asked for, rather than when we receive a request

Reply:
Sure.